rotkehlchen/db/accounting_rules.py

```diff
@@ -336,3 +303,151 @@ def query_rules_and_serialize(
                 data[linked_property]['linked_setting'] = setting_name
             entries.append(data)
         return entries, total_found_result
+
+
+def _check_if_event_affected_by_treatment(
+        cursor: 'DBCursor',
+        callbacks: dict[int, tuple[int, EventsAccountantCallback]],
+        related_events: Sequence[HistoryBaseEntry],
+        event: HistoryBaseEntry,
+        pot: 'AccountingPot',
+) -> bool:
+    """Check if an event will be affected by a callback or treatment from a previous event"""
+    for related_event_idx, related_event in enumerate(related_events):
+        counterparty = getattr(related_event, 'counterparty', None)
+        if counterparty == CPT_GAS:  # avoid checking the case of gas in evm events
+            continue
+
+        # query the accounting rule for the related event
+        cursor.execute(
+            'SELECT accounting_treatment FROM accounting_rules WHERE (type=? AND subtype=? AND (counterparty=? OR counterparty=?))',  # noqa: E501
+            (
+                related_event.event_type.serialize(),
+                related_event.event_subtype.serialize(),
+                NO_ACCOUNTING_COUNTERPARTY if counterparty is None else counterparty,
+                NO_ACCOUNTING_COUNTERPARTY,
+            ),
+        )
+
+        if (raw_treatment := cursor.fetchone()) is not None and raw_treatment[0] is not None:
+            accounting_treatment = TxAccountingTreatment.deserialize_from_db(raw_treatment[0])
+            match accounting_treatment:
+                case TxAccountingTreatment.SWAP:
+                    # ensure that the event that we are going to check can exist
+                    # in the related events. Avoid index error
+                    if len(related_events) <= related_event_idx + 1:
```

Comment:
read what I wrote in the last review for this: https://github.com/rotki/rotki/pull/6953#discussion_r1399607695